### ISSUE TITLE
Building on Linux

### ISSUE THREAD

**Issue by brentnd** [Repkap11/RepGame#7]
When building and running, this is what I see:

![image](https://user-images.githubusercontent.com/4182027/55349471-cf7e7480-5487-11e9-9b22-fb320085d77d.png)

The sky moves around with mouse movement but the window cannot be expanded.

**Commands:**
```
sudo make install
make linux
./RepGame
```
**Environment:**
* Branch: master
* Ubuntu 18.04
* gcc 7.3.0
* freeglut 2.8.1-3
* glew 2.0.0-5
* glm 0.9.9~a2-2
* nvidia-cuda-toolkit 9.1.85-3ubuntu1
* rsync 3.1.2-2.1ubuntu1

Any ideas?

**Comment by Repkap11** [comment]
If you type "i" it should open your inventory, free your mouse, and let you
maximize the window.
I'm not sure why you can't see the world...

**Comment by brentnd** [comment]
Cool, "i" helped me get to what looks like a broken inventory and maximize the window.

![image](https://user-images.githubusercontent.com/4182027/55355031-335b6a00-5495-11e9-955b-670e2350259a.png)

No notable logs or errors in the console output during that. Seems like I'm maybe missing resources or sprites.

**Comment by brentnd** [comment]
Could it be something with this?

https://github.com/Repkap11/RepGame/blob/master/src/common/abstract/textures.cpp#L65

Do I need to symlink the bitmaps to `/system/bin` or something?

**Comment by brentnd** [comment]
Nevermind, I guess that's just for caching the textures.

**Comment by Repkap11** [comment]
That path is for Android, which has strange paths. The image should be this one. 
https://github.com/Repkap11/RepGame/blob/master/bitmaps/textures.bmp
Its looked for relative to the executable as bitmaps/textures.bmp

**Comment by brentnd** [comment]
Yeah, I did some debugging and `texture_init_blocks` and `texture_init_sky` are both resolving the correct path for the bitmaps.

**Comment by Repkap11** [comment]
Could be your GPU doesn't support the texture mode I'm using. Do you get black ground if you comment out the "discard" in chunk_fragment.glsl?

**Comment by brentnd** [comment]
I commented out each "discard" individually and all of them together and didn't see any difference in the ground color.

**Comment by Repkap11** [comment]
Ahh, comment out the #include cuda.mk inside linux.mk
I didn't realize the install instructions will let you install the nvcc compiler even if the host doesn't support cuda acceleration.

**Comment by brentnd** [comment]
Okay, with cuda commented out and rebuilt, it starts like it did before then the screen goes back around 5 seconds. Inventory is still a bunch of black boxes.

Something I just thought of is that I am running this on a vagrant VM with virtualbox. Could that be problematic?

**Comment by Repkap11** [comment]
Try "make map" to delete your "World1" directory.
I still can't explain the blackness.
It looks like your VM has OpenGL 3.3 core, which is what it needs.

**Comment by brentnd** [comment]
Okay, seems a bit better now. I can move around and see black blocks where the world has actually generated ground (instead of just the while abyss from before). Still no textures on inventory either so seems like that's what remains. Could be lack of graphic resources on my VM.
